### DISCUSSION TITLE
MACHINE definition for orin nx

### DISCUSSION THREAD

**Discussion by bchoineubility** [OE4T/tegra-demo-distro#296] [Q&A]
Hi,

When I try to build yocto with orin nx, what is MACHINE for it?

I can see machines you sampled in the local.conf.sample file.

```
#MACHINE ?= "jetson-agx-orin-devkit"
#MACHINE ?= "jetson-agx-xavier-devkit"
#MACHINE ?= "jetson-xavier-nx-devkit"
#MACHINE ?= "jetson-xavier-nx-devkit-emmc"
```

BR,
Mark

**Comment by dwalkes** [comment]
Hi Mark
See https://docs.nvidia.com/jetson/archives/r35.5.0/DeveloperGuide/IN/QuickStart.html#jetson-modules-and-configurations and the machines defined in https://github.com/OE4T/meta-tegra/tree/master/conf/machine

**Reply by bchoineubility** [reply]
In my case, is it right using jetson-orin-nano-devkit for machine?

Besides my question, can you let me know schedule when Jetpack 6.0 is available for yocto?

**Reply by dwalkes** [reply]
>  it right using jetson-orin-nano-devkit for machine?

Probably not, but you'd need to cross reference the specific orin NX hardware setup you have with the links I sent above.  For instance if you had the 16GB with P3509 carrier you'd want [p3509-a02-p3767-0000](https://github.com/OE4T/meta-tegra/blob/master/conf/machine/p3509-a02-p3767-0000.conf#L3) I believe.

**Comment by ichergui** [comment]
Hi @bchoineubility 
`JetPack 6.0 / L4T R36.2.0 DP` is available. 
Please check our [WIP](https://github.com/OE4T/meta-tegra-community/tree/wip-l4t-r36.2.0-dp) branch.

**Reply by bchoineubility** [reply]
Hi @ichergui 
Thanks for reply.

Can it be built combination with Kirkstone ?

**Reply by dwalkes** [reply]
> Can it be built combination with Kirkstone ?

No, and there are no plans to support a kirkstone Jetpack 6 branch at this time, based on the [discussion in the Feb meeting](https://github.com/OE4T/meta-tegra/wiki/OE4T-Meeting-Notes-2024-02-08).  If this is something you need you consider [attending the next meeting](https://github.com/OE4T/meta-tegra/discussions/515#discussioncomment-8887272) and/or volunteering to help implement.

**Reply by bchoineubility** [reply]
Hi @dwalkes 

If I try to build with JP6, should I use Scarthgap branch for poky and other meta layers?
If so, I am not sure other meta-layers are ready for Scarthgap especially ROS2.
